pyproject.toml

```
@@ -42,7 +42,6 @@ dependencies = [
     "ipaddress; python_version<'3.7'",
     "opentelemetry-api>=1; python_version>='3.7'",
     "pathlib2; python_version<'3.5'",
-    "pep562; python_version<'3.7'",
     "protobuf>=3,<3.18; python_version<'3.6'",
     "protobuf>=3,<4.0; python_version=='3.6'",
     "protobuf>=3; python_version>='3.7'",
```

Not sure if this in the scope of this PR but we can simplify the dependencies required (drop all references to py<=3.7).

```suggestion
    "protobuf>=3",
```

---

Yup, this is done in #6626 here: https://github.com/DataDog/dd-trace-py/pull/6626/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711